Content.Shared/Clothing/LoadoutSystem.cs

```diff
@@ -145,6 +145,39 @@ private void OnMapInit(EntityUid uid, LoadoutComponent component, MapInitEvent a
         Equip(uid, component.StartingGear, component.RoleLoadout);
     }
 
+    // Moffstation - Begin - Custom function for Equipping which allows for antag roles to make use of loadouts
+    public void LoadoutAwareEquip(EntityUid uid,
+        ICommonSession session,
+        List<ProtoId<StartingGearPrototype>>? startingGear,
+        List<ProtoId<RoleLoadoutPrototype>>? loadoutGroups,
+        HumanoidCharacterProfile profile)
+    {
+        if (startingGear != null && startingGear.Count > 0)
+            _station.EquipStartingGear(uid, _random.Pick(startingGear));
+
+        if (loadoutGroups != null && loadoutGroups.Count > 0)
+        {
+            foreach (var antagLoadout in loadoutGroups)
+            {
+                RoleLoadout? loadout = null;
+                if (_protoMan.TryIndex(antagLoadout, out RoleLoadoutPrototype? roleProto))
+                {
+                    profile?.Loadouts.TryGetValue(antagLoadout, out loadout);
+
+                    if (loadout == null)
```

Comment:
```suggestion
                    if (!profile.Loadouts.TryGetValue(antagLoadout, out var loadout))
```
(And you need to get rid of `RoleLoadout? loadout = null;` higher up)

`profile`'s not nullable in this scope, so this can be dramatically simplified.